crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -47,6 +47,16 @@ impl Persistence {
             .map_err(Error::DbError)
     }
 
+    pub async fn solvable_orders(
+        &self,
+        min_valid_to: u32,
+    ) -> Result<domain::SolvableOrders, Error> {
```

Comment:
`Result<boundary::SolvableOrders, Error>` so that we respect the ddd rules

Reply:
Wouldn't DDD actually expect that the infra returns the domain type?
The `domain` should never have to cater to anyone else's needs and everything else should work with whatever the `domain` hands it or return whatever the `domain` expects.
That way the `domain` does not become verbose by having to convert between the different types.

Reply:
Yes, this comment is a follow up for this one https://github.com/cowprotocol/services/pull/2314#discussion_r1462448013

Reply:
Moved the struct to `boundary` for now